Assets/_Scripts/_Event System/GameEventSystem.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GameEventSystem : MonoBehaviour
+{
+    private readonly Dictionary<Type, Delegate> events = new();
+
+    public static GameEventSystem Instance;
+
+    private void Awake()
+    {
+        if (Instance == null)
+        {
+            Instance = this;
+        }
+        else
+        {
+            Destroy(gameObject);
+        }
+    }
+
+    // Raise method for events that do NOT return a result (Action<TEvent>)
+    public void Raise<TEvent>(TEvent eventObject)
+    {
+        if (events.TryGetValue(typeof(TEvent), out Delegate existing))
+        {
+            if (existing is Action<TEvent> action)
+            {
+                action.Invoke(eventObject);
+            }
+            else
+            {
+                throw new InvalidOperationException(
+                    $"TEvent {typeof(TEvent)} is expected to use a Func, but a different delegate type is registered."
+                );
+            }
+        }
+        else
+        {
+            Debug.LogWarning("Event not found!");
+        }
+    }
+
+    // Raise method for events that do not return a result (Func<TEvent, TResult>)
```

Comment:
Wrong comment 😅  should be:

`// Raise method for events that return a result (Func<TEvent, TResult>)`